recipe/build.sh

```diff
@@ -5,51 +5,51 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 10.* ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.0* ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.1 ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.2 ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 11.8 ]]; then
-      export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
-      export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
+    export TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9+PTX"
+    export CUDA_TOOLKIT_ROOT_DIR=$CUDA_HOME
   elif [[ ${cuda_compiler_version} == 12.0 ]]; then
-      export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
-      # $CUDA_HOME not set in CUDA 12.0. Using $PREFIX
-      export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}"
+    export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
+    # $CUDA_HOME not set in CUDA 12.0. Using $PREFIX
+    export CUDA_TOOLKIT_ROOT_DIR="${PREFIX}"
   else
-      echo "unsupported cuda version. edit build.sh"
-      exit 1
+    echo "unsupported cuda version. edit build.sh"
+    exit 1
   fi
 
 else
   export FORCE_CUDA=0
 fi
 
-if [[ "${target_platform}" == osx-* ]]; then
-  export CXXFLAGS="-D_LIBCPP_DISABLE_AVAILABILITY ${CXXFLAGS}"
-fi
+# Dynamic libraries need to be lazily loaded so that torch
+# can be imported on system without a GPU
+export LDFLAGS="${LDFLAGS//-Wl,-z,now/-Wl,-z,lazy}"
 
 # export USE_MKL_BLAS=1  # only used for >0.1.0
-export Torch_DIR=$(python -c 'import torch; print(torch.utils.cmake_prefix_path)')
-
 export FORCE_NINJA=1
 export EXTERNAL_PHMAP_INCLUDE_DIR="${BUILD_PREFIX}/include/"
 export EXTERNAL_CUTLASS_INCLUDE_DIR="${BUILD_PREFIX}/include/"
 
-python -m pip install . -vvv
+export PYG_CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${PREFIX} -DTorch_DIR=${PREFIX}/lib/python${PY_VER}/site-packages/torch/share/cmake/Torch"
```

Comment:
I think it would be good to leave a comment that this variable is directly related to the patch:
```suggestion
# note that we patch PYG_CMAKE_ARGS into setup.py, in order to set Torch_DIR correctly also for cross-compilation
export PYG_CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_INSTALL_PREFIX=${PREFIX} -DTorch_DIR=${PREFIX}/lib/python${PY_VER}/site-packages/torch/share/cmake/Torch"
```